test/data/project_detection/TestPackage2/Project.toml

```diff
@@ -0,0 +1,22 @@
+name = "TestPackage2"
+uuid = "acc75194-7866-4657-b784-337cf9e0ca80"
+authors = ["David Anthoff <anthoff@berkeley.edu>"]
+version = "1.0.0"
+
+[deps]
+TestItems = "1c621080-faea-4a02-84b6-bbd5e436b8fe"
+JuliaSyntax = "70703baa-626e-46a2-a12c-08ffd08c73b4"
+Pkg = "44cfe95a-1eb2-52ea-b672-e2afdf69b78f"
+UUIDs = "cf7118a7-6976-5b1a-9a39-7adc72f591a4"
+
+[extras]
+TestItemRunner = "f8b46487-2199-4994-9208-9a1283c18c0a"
+Test = "8dfed614-e22c-5e08-85e1-65c5234f0b40"
+
+[compat]
+julia = "1"
+JuliaSyntax = "0.3"
+TestItems = "0.1"
+
+[targets]
+test = ["Test", "TestItemRunner"]
```

Comment:
Can you add a test case with a Manifest.toml, and make sure that when the tests are run, the manifest is respected?

That is the specific case we originally ran into in our issue.

One easy way to have an independent Manifest.toml is to have a subpackage that is dev'ed from the higher-level package, which would only be reflected in the Manifest. So if you aren't correctly respecting it, the tests will complain that the package isn't registered and can't be found. (that's what we encountered.)

Reply:
(thanks for merging this and pulling it forward!) :)

Reply:
There is a test case with a `Manifest.toml`, but the `Manifest.toml` is not checked in because then we run into all the compat problems across Julia versions and the test wouldn't run on different Julia versions. Instead the `Manifest.toml` gets created [here](https://github.com/julia-vscode/JuliaWorkspaces.jl/blob/bcc6a553d603e225f6340dc184dc39dfc1688443/test/test_testitems.jl#L240) and then the tests below make sure that is properly detected.

> when the tests are run

There is no logic in this package to run anything, so all the test can cover is that the manifest gets properly detected, which this does. We could probably add a test case to TIR2.jl to have a test that makes sure it is all correctly done when running tests.

Reply:
Thanks, that makes a ton of sense!

💡 Cool, thanks for doing that!